src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -253,7 +253,25 @@ public sealed class ActivityListener : IDisposable
         public System.Diagnostics.SampleActivity<string>? SampleUsingParentId { get { throw null; } set { throw null; } }
         public System.Diagnostics.SampleActivity<ActivityContext>? Sample { get { throw null; } set { throw null; } }
         public void Dispose() { throw null; }
-   }
+    }
+
+    public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
```

Comment:
This could be nested inside `TextMapPropagator.GetterCallback` / `TextMapPropagator.SetterCallback` if there is a potential for something other than `TextMap*` to exist in the future.

Should the setter have its own custom delegate too to match the getter (I know it's not technically required as with the getter)?

Reply:
Sounds a good idea.